### ISSUE TITLE
--git flag misattributes lint errors to wrong files

### ISSUE THREAD

**Issue by privatenumber** [privatenumber/lintroll#107]
## Error

```
tests/stress/stress-errors.js
   93:3  error    Use `switch` instead of multiple `else-if`  unicorn/prefer-switch
```

## Problem

When using `--git`, lint errors can be attributed to the wrong file. In this case, the `unicorn/prefer-switch` error at line 93 belongs to `stress-max-visible.js` (which has an `else-if` chain at line 93), but it was reported against `stress-errors.js` (where line 93 is a string literal `'Level 2 (fails)'`).

### Reproduction

In a repo where `git ls-files` returns both `examples/` and `tests/stress/` files (the stress files were moved from examples via git):

```bash
# Without --git: error correctly attributed to stress-max-visible.js
npx lintroll --node --no-cache tests/stress/
# stress-max-visible.js:93:3 error unicorn/prefer-switch ✓

# With --git: error incorrectly attributed to stress-errors.js
npx lintroll --node --no-cache --git tests/stress/
# stress-errors.js:93:3 error unicorn/prefer-switch ✗
```

Running on a single file also showed the discrepancy:

```bash
# Without --git: 0 errors (correct)
npx lintroll --node --no-cache tests/stress/stress-errors.js

# With --git: 1 error at line 93 (incorrect - no else-if exists in this file)
npx lintroll --node --no-cache --git tests/stress/stress-errors.js
```

### Context

- The `--git` flag uses `git ls-files` to determine which files to lint
- Both old (`examples/`) and new (`tests/stress/`) paths were tracked by git
- The error message referenced code that only exists in a different file (`stress-max-visible.js`), not in the reported file (`stress-errors.js`)

### Root Cause Hypothesis

The `--git` flag changes the set of files passed to `eslint.lintFiles()` compared to the default directory scan. When both the original and moved copies of files are tracked by git, ESLint may process both sets and misattribute results between files that happen to share the same line numbers.

This could also be an ESLint caching issue where cached results from one file path are applied to a different file path when the file set changes.

### Suggested Investigation

1. Check if `git ls-files` returning both old and new paths (after a rename) causes file list overlap
2. Consider whether `--git` should use `git ls-files --deduplicate` or filter out files that have rename counterparts
3. Test if disabling ESLint's internal parallel processing resolves the misattribution

**Comment by privatenumber** [comment]
## Additional context

The misattribution happened specifically in CI (GitHub Actions), not locally with the same `--no-cache` flag. Here's what makes this tricky:

### Environment
- CI runner: `ubuntu-latest` on GitHub Actions
- Node: installed via setup-node
- The lint command: `lintroll --node --cache --git`
- CWD = gitRoot (repo root)

### Git state when it happened
The files were moved in a single commit via `git mv`:
```
renamed: examples/stress-errors.js -> tests/stress/stress-errors.js
renamed: examples/stress-max-visible.js -> tests/stress/stress-max-visible.js
(+ 3 more renames)
```

The `--git` flag compares against the base branch (`master`), so both old and new paths may appear in the diff.

### The specific error
```
tests/stress/stress-errors.js
   93:3  error    Expected a default case       default-case
```

But `stress-errors.js:93` is a string literal (`'Level 2 (fails)'`). The `default-case` error actually belongs to `stress-max-visible.js` which had a `switch(0)` without a `default` case — and that file's line 93 is exactly where the switch was.

### Why I think it's related to the `--git` flag
- `npx lintroll --node --no-cache tests/stress/stress-errors.js` → 0 errors ✓
- `npx lintroll --node --no-cache --git tests/stress/stress-errors.js` → 1 error (the phantom)

But only reproducible intermittently. After running `--fix` + recommitting, the error persisted in CI but not locally (even with `--no-cache`). It's possible the `--git` flag resolved a stale file list from the rename diff that caused ESLint to process files in an unexpected order.

### Could not reproduce after
Once we moved the files out of git tracking entirely (to `.local/scripts/` + `.gitignore`), the error disappeared. So we can't easily reproduce anymore.

The `git ls-files` path resolution bug you found could explain this if CI was running in a context where paths resolved differently than expected.

**Comment by privatenumber** [comment]
CI doesn't persist `.eslintcache` — there's no `actions/cache` for it. Each CI run starts fresh. So cache corruption from a previous run isn't the cause.

The `--cache` flag in the lint command creates a cache within the same run, but the rename happened in a prior commit — by the time CI runs, `git ls-files` only sees the new paths as you said.

Given that, I'm not sure what's left to explain the misattribution. It may just be a rare edge case that's hard to reproduce. Happy to close this if there's nothing actionable — the `--full-name` fix in #109 is a good improvement regardless.

**Comment by privatenumber** [comment]
Closing — the misattribution isn't reproducible and the root cause remains unclear. The `--full-name` fix in #109 is a real improvement to path resolution that could prevent similar issues.
